docs/examples/index.md

```diff
@@ -54,14 +215,104 @@ object Uppercase extends IOApp.Simple:
     yield ()
 
 end Uppercase
+```
+
+@:choice(static)
+
+```scala 3 mdoc:reset
+import cats.effect.{IO, IOApp}
+import fs2.io.file.Path
+
+import shellfish.os.syntax.FilesOs
+
+object Uppercase extends IOApp.Simple:
+
+  val path      = Path("src/main/resources/quijote.txt")
+  val upperPath = Path("src/main/resources/quijote_screaming.txt")
+
+  def run: IO[Unit] =
+    for
+      file <- FilesOs.read(path)
+      _    <- FilesOs.write(upperPath, file.toUpperCase)
+    yield ()
+
+end Uppercase
+```
+
+@:choice(fs2)
+
+```scala 3 mdoc:reset
+import cats.effect.{IO, IOApp}
+
+import fs2.io.file.{Path, Files}
+
+
+object Uppercase extends IOApp.Simple:
+
+  val path      = Path("src/main/resources/quijote.txt")
+  val upperPath = Path("src/main/resources/quijote_screaming.txt")
+
+  def run: IO[Unit] =
+    Files[IO].readUtf8(path)
+      .map(_.toUpperCase)
+      .through(Files[IO].writeUtf8(upperPath))
+      .compile
+      .drain
 
+end Uppercase
 ```
 
+@:@
+
 ## Places
 
-Shellfish OS has compatibility for working with [scodec](https://scodec.org/)!
+Maybe you also want to work with binary files. In this example, we are going to create a binary file with a list of places. Because Shellfish has compatibility for working with [scodec](https://scodec.org/), we are going to use that library to encode and decode the binary file!
 
-``` scala 3 mdoc
+As always, we start by defining type representations of the data we are going to work with:
+
+```scala 3
+case class Place(position: Int, contestant: String)
+```
+
+After that, we are going to use the `scodec` library to create a codec for the `Place` type.
+
+A first approach would be to create your own custom codec using the combinators of scodec:
+
+```scala 3
+import scodec.codecs.*
+
+given placeCodec: Codec[Place] = (uint8 :: utf8).as[Place]
+```
+
+But since Scala 3, you can also use the `derives` feature of Scala 3 to automatically derive a codec for your type:
+
+```scala 3
+case class Place(position: Int, contestant: String) derives Codec
+```
+That will automatically create a given instance of `Codec[Place]` for you!
+
+Then, we can create a file with a list of places, read it, and print it to the console:
+
+```scala 3
+for
+    exists <- path.exists                                       // (1)
+    _      <- path.createFile.whenA(exists)                     // (2)
```

Comment:
Isn't this exactly the opposite? You have to create the file if it doesn't exist! (so it's `unlessA`)
Also `path.createFile` already creates the file: are you doing this just check because `createFile` throws exception when the file already exists?

Reply:
Yup, its because of the exception!